app/models/semester.rb

```diff
@@ -1,29 +1,32 @@
+# frozen_string_literal: true
+
 class Semester < ActiveRecord::Base
   include Authority::Abilities
   has_many :freeform_shows, dependent: :destroy
   has_many :specialty_shows, dependent: :destroy
   has_many :talk_shows, dependent: :destroy
 
+  before_validation :ensure_beginning_and_ending_are_at_six_am
+
   validates :beginning, :ending, presence: true
-  validate :semesters_dont_overlap
-  validate :beginning_before_ending
+  validate :discrete_semester_dates
 
-  before_save :ensure_beginning_and_ending_are_at_six_am
   after_create { Signoff.propagate_all(beginning, ending) }
 
   default_scope { order(beginning: :desc) }
 
-  def semesters_dont_overlap
-    conflicts = Semester.all.select{ |s| (beginning - s.ending) * (s.beginning - ending) > 0 }
+  def discrete_semester_dates
+    if beginning > ending
+      errors.add(:ending, 'must be later than start date')
+      return
+    end
 
+    conflicts = Semester.where(
+      'tsrange(beginning, ending) && tsrange(?, ?)',
+      beginning, ending
+    )
     return unless conflicts.any?
```

Comment:
Can you verify for me (by looking in the log when you run the page) that even separated into two statements the generated SQL is `select 1 as one from...` I think it should be, since AR queries are lazy, but let's just double check